floodlight/core/events.py

```diff
@@ -389,3 +391,106 @@ def rotate(self, alpha: float):
                 self.events[["to_x", "to_y"]] = pd.DataFrame(
                     np.round(np.dot(self.events[["to_x", "to_y"]], r), 3)
                 )
+
+    def slice(
+        self,
+        start: float = None,
+        end: float = None,
+        slice_by="gameclock",
+        inplace: bool = False,
+    ):
+        """Return copy of object with events sliced in a time interval using either the
+        gameclock (total seconds) or the frameclock. All entries without a valid time
+        entry (e.g. None) are excluded.
+
+        Parameters
+        ----------
+        start : float, optional
+            Start frame or second of slice. Defaults to beginning of segment.
+        end : float, optional
+            End frame or second of slice (endframe is excluded). Defaults to last event
+            of segment (including).
+        slice_by: str, optional
+            Column used to slice the events. Defaults to ``gameclock``.
+        inplace: bool, optional
+            If set to ``False`` (default), a new object is returned, otherwise the
+            operation is performed in place on the called object.
+
+        Returns
+        -------
+        events_sliced: Union[Event, None]
+        """
+        if slice_by not in self.events:
+            ValueError(f"Events object does not contain column {slice_by}!")
+        if start is None:
+            start = 0
+        if end is None:
+            end = np.nanmax(self.events[slice_by].values) + 1
+
+        sliced_data = self.events[self.events[slice_by] >= start].copy()
+        sliced_data = sliced_data[sliced_data[slice_by] < end]
+
+        events_sliced = None
+        if inplace:
+            self.events = sliced_data
+        else:
+            events_sliced = Events(
+                events=sliced_data,
+                direction=deepcopy(self.direction),
+            )
+
+        return events_sliced
+
+    def get_event_stream(
+        self,
+        fade: int = 0,
+        **kwargs,
+    ) -> Code:
+        """Generates a continuous Code object containing the eIDs of all events at the
+        corresponding frame numbers. The eID are maintained for a certain fade duration.
+        Remaining values are set to np.nan. Requires the protected column ``frameclock``
+        to function.
+
+        Parameters
+        ----------
+        fade: int, optional
+            Number of additional frames for which the Code object should stay at a
+            value after the event occurred. The value is overwritten if another event
+            occurs within the fade duration. If chosen to zero, the value is maintained
+            only for a single frame. If chosen to None, the value is maintained until
+            either the next event or until the end of the sequence. Defaults to 0.
+        kwargs:
+            Other keyword arguments ("name", "definitions", "framerate") are passed down
+             to the Code object.
```

Comment:
That's unclear to me

Reply:
Ok, I specified this part to be more explicit about what's happening. 
But I still think these are useful to simplify the generation of a code object, e.g.

possession  = events.get_event_stream(name="possession", definitions={0: "Home", 1: "Away"}